website/project/views/node.py

```diff
@@ -1008,6 +1008,66 @@ def get_children(auth, node, **kwargs):
     return _render_nodes(nodes, auth)
 
 
+def node_child_tree(user, node_ids):
+    """ Format data to test for node privacy settings for use in treebeard.
+    """
+    items = []
+    addons = []
+    for node_id in node_ids:
+        node = Node.load(node_id)
+        assert node, '{} is not a valid Node.'.format(node_id)
+
+        can_read = node.has_permission(user, 'read')
+        can_read_children = node.has_permission_on_children(user, 'read')
+        if not can_read and not can_read_children:
+            continue
+        for addon in node.get_addon_names():
+            if (addon is not 'osfstorage') and (addon is not 'wiki') and (addons.count(addon) != 1):
+                addons.append(addon)
+        children = []
+        # List project/node if user has at least 'read' permissions (contributor or admin viewer) or if
+        # user is contributor on a component of the project/node
+        can_write = node.has_permission(user, 'admin')
+        children.extend(node_child_tree(
+            user,
+            [
+                n._id
+                for n in node.nodes
+                if n.primary and
+                not n.is_deleted
+            ]
+        ))
+        item = {
+            'node': {
+                'id': node_id,
+                'url': node.url if can_read else '',
+                'title': node.title if can_read else 'Private Project',
+                'is_public': node.is_public,
+                'addons': addons,
+                'can_write': can_write
+            },
+            'user_id': user._id,
+            'children': children,
+            'kind': 'folder' if not node.node__parent or not node.parent_node.has_permission(user, 'read') else 'node',
+            'nodeType': node.project_or_component,
+            'category': node.category,
+            'permissions': {
+                'view': can_read,
+            }
+        }
+
+        items.append(item)
+
+    return items
+
+
+@must_be_logged_in
+@must_be_valid_project
+def get_node_tree(auth, **kwargs):
+    node = kwargs.get('node') or kwargs['project']
```

Comment:
Minor: Pulling off of `kwarg` is deprecated. Explicitly add `node` to the function signature instead

```python
def get_node_tree(auth, node **kwargs):
    #...
```

Btw, I just saw that COSDev was out of date w/r/t this. I just updated them.

Anyway, this doesn't need to block merging.